src/java/org/apache/cassandra/index/sai/StorageAttachedIndex.java

```diff
@@ -610,6 +610,7 @@ public void insertRow(Row row)
         @Override
         public void updateRow(Row oldRow, Row newRow)
         {
+            indexContext.unindex(key, oldRow, mt, CassandraWriteContext.fromContext(writeContext).getGroup());
```

Comment:
there will be a race condition when updating non-vector column:  existing vector will not be queriable until it's re-inserted

Reply:
We need to check if vector column value has changed before unindex